lutra/bindings/python/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+use arrow::{pyarrow::PyArrowType, record_batch::RecordBatch};
+use itertools::Itertools;
+use std::str::FromStr;
+
+use pyo3::prelude::*;
+
+#[pymodule]
+fn lutra(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(execute_one, m)?)?;
+    // From https://github.com/PyO3/maturin/issues/100
+    m.add("__version__", env!("CARGO_PKG_VERSION"))?;
+    Ok(())
+}
+
+#[pyfunction]
+pub fn execute_one(
+    project_path: &str,
+    expression_path: &str,
+) -> PyResult<PyArrowType<RecordBatch>> {
```

Comment:
As you may know, this package probably does not need to have `pyarrow` as a dependency for this function.
It should be sufficient to implement The Arrow PyCapsule Interface.
https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html

Reply:
Oh, I see - it's a wrapper for a pointer to a PyArrow object, that is supposed to be arrow version independent.

And that makes it able to pass it to Python as a PyCapsule, so pyarrow dep is not needed.

Reply:
https://github.com/apache/arrow-rs/issues/5079

https://github.com/apache/arrow-rs/pull/5070

> It's not possible (I don't think) to implement exporting to the PyCapsule interface because no Python classes are exposed.

It looks like we cannot do this yet. When arrow-rs changes its implementation, I think we'll automatically start producing PyCapsules, so that's nice.

Reply:
This interface must be implemented in a Python class and probably will not be provided by arrow-rs itself?

In any case, we can take a look at it later.